### PR TITLE
PROFIN-27 | agrego filtro de deliveries por disponibilidad

### PR DESCRIPTION
**Pull request by MarcosPereyra**

### CODE REVIEW

**Comment by code-race-2021** [comment]
# [Codecov](https://codecov.io/gh/code-race-2021/final-project-cruds/pull/9?src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None) Report
> Merging [#9](https://codecov.io/gh/code-race-2021/final-project-cruds/pull/9?src=pr&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None) (4f1727b) into [master](https://codecov.io/gh/code-race-2021/final-project-cruds/commit/d0932988c0449cf8325e2d5fef8759b8d211bd15?el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None) (d093298) will **decrease** coverage by `0.24%`.
> The diff coverage is `88.88%`.

[Impacted file tree graph](https://codecov.io/gh/code-race-2021/final-project-cruds/pull/9?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None)

```diff
@@             Coverage Diff              @@
##             master       #9      +/-   ##
============================================
- Coverage     96.17%   95.93%   -0.25%     
- Complexity      168      171       +3     
============================================
  Files            26       26              
  Lines           314      320       +6     
  Branches          4        5       +1     
============================================
+ Hits            302      307       +5     
  Misses            4        4              
- Partials          8        9       +1     
```


| [Impacted Files](https://codecov.io/gh/code-race-2021/final-project-cruds/pull/9?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None) | Coverage Δ | |
|---|---|---|
| [...ain/java/com/coderace/service/DeliveryService.java](https://codecov.io/gh/code-race-2021/final-project-cruds/pull/9/diff?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None#diff-c3JjL21haW4vamF2YS9jb20vY29kZXJhY2Uvc2VydmljZS9EZWxpdmVyeVNlcnZpY2UuamF2YQ==) | `96.42% <85.71%> (-3.58%)` | :arrow_down: |
| [...va/com/coderace/controller/DeliveryController.java](https://codecov.io/gh/code-race-2021/final-project-cruds/pull/9/diff?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None#diff-c3JjL21haW4vamF2YS9jb20vY29kZXJhY2UvY29udHJvbGxlci9EZWxpdmVyeUNvbnRyb2xsZXIuamF2YQ==) | `100.00% <100.00%> (ø)` | |
| [...main/java/com/coderace/service/ExampleService.java](https://codecov.io/gh/code-race-2021/final-project-cruds/pull/9/diff?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None#diff-c3JjL21haW4vamF2YS9jb20vY29kZXJhY2Uvc2VydmljZS9FeGFtcGxlU2VydmljZS5qYXZh) | `100.00% <100.00%> (ø)` | |

------

[Continue to review full report at Codecov](https://codecov.io/gh/code-race-2021/final-project-cruds/pull/9?src=pr&el=continue&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/code-race-2021/final-project-cruds/pull/9?src=pr&el=footer&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None). Last update [d093298...4f1727b](https://codecov.io/gh/code-race-2021/final-project-cruds/pull/9?src=pr&el=lastupdated&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None).

File: src/main/java/com/coderace/controller/DeliveryController.java [hunk 1]
```diff
@@ -28,8 +30,7 @@
     }
 
     @GetMapping
-    public ResponseEntity<Object> getAll() {
-        return ResponseEntity.ok().body(this.service.getAll());
+    public ResponseEntity<Object> getAll(@RequestParam (required = false) boolean available) {
+        return ResponseEntity.ok().body(this.service.getAll(available));
```
Comment (NeiMeli) [inline src/main/java/com/coderace/controller/DeliveryController.java]: Agregaría default value false por si mandan nulo:
`(required = false, defaultValue = "false")`

File: src/main/java/com/coderace/service/DeliveryService.java [hunk 2]
```diff
@@ -30,8 +31,19 @@ public DeliveryResponseDTO create(DeliveryRequestDTO requestDTO) {
...
+                    .map(this::buildDeliveryResponseDTO)
+                    .collect(Collectors.toList());
+    }
+
+    private boolean isAvailable(LocalDateTime date, boolean available) {
+        if (available) {        // si available es true...
+            return date == null;    // devuelvo sólo los date que son null
+        } else {
+            return true;        // si available es false, devuelvo todos los valores por default
+        }
```
Comment (NeiMeli) [inline src/main/java/com/coderace/service/DeliveryService.java]: Creo que esto se puede simplificar en una línea:
_Devolver true si available es false o sino, ya que available es true, comparar el date con null_

File: src/test/java/com/coderace/service/DeliveryServiceTest.java [hunk 3]
```diff
@@ -98,20 +98,34 @@ void buildDeliveryResponseDTO() {
...
         final List<Delivery> all = Collections.singletonList(delivery);
 
         when(repository.findAll()).thenReturn(all);
 
-        final List<DeliveryResponseDTO> result = service.getAll();
+        final List<DeliveryResponseDTO> result = service.getAll(true);
 
         assertEquals(1, result.size());
         assertEquals(service.buildDeliveryResponseDTO(delivery), result.get(0));
     }
```
Comment (NeiMeli) [inline src/test/java/com/coderace/service/DeliveryServiceTest.java]: Este test está bien, pero agregaría a la lista 'all' un delivery cuya fecha NO sea nula, así nos aseguramos que el filtro lo ignore. De paso hacemos que pase el codecov, ya que está indicando que en ningún test se pasa por la línea 43 del service `return date == null` y que la comparación de false.

**Comment by MarcosPereyra** [comment]
Nei, ya corregí los tests de getAll()

**Review by NeiMeli — APPROVED** [review]
Excelente!

**Pull request merged**